src/devices/Mcp25xxx/samples/Mcp25xxx.Sample.cs

```diff
@@ -0,0 +1,654 @@
+
+            using (Mcp25xxx mcp25xxx = GetMcp25xxxDevice())
+            {
+                Reset(mcp25xxx);
+
+                // ReadAllRegisters(mcp25xxx);
+                ReadAllRegistersWithDetails(mcp25xxx);
+                // ReadRxBuffer(mcp25xxx);
+                // Write(mcp25xxx);
+                // LoadTxBuffer(mcp25xxx);
+                // RequestToSend(mcp25xxx);
+                // ReadStatus(mcp25xxx);
+                // RxStatus(mcp25xxx);
+                // BitModify(mcp25xxx);
+                // TransmitMessage(mcp25xxx);
+                //LoopbackMode(mcp25xxx);
+                //ReadAllRegisters(mcp25xxx);
+            }
+        }
+
+        private static Mcp25xxx GetMcp25xxxDevice()
+        {
+            var spiConnectionSettings = new SpiConnectionSettings(0, 0);
+            var spiDevice = new UnixSpiDevice(spiConnectionSettings);
+            return new Mcp25625(spiDevice);
+        }
+
+        private static void Reset(Mcp25xxx mcp25xxx)
+        {
+            Console.WriteLine("Reset Instruction");
+            mcp25xxx.Reset();
+        }
+
+        private static void ReadAllRegisters(Mcp25xxx mcp25xxx)
+        {
+            Console.WriteLine("Read Instruction for All Registers");
+            Array addresses = Enum.GetValues(typeof(Address));
+
+            foreach (Address address in addresses)
+            {
+                byte addressData = mcp25xxx.Read(address);
+                Console.WriteLine($"0x{(byte)address:X2} - {address,-10}: 0x{addressData:X2}");
+            }
+        }
+
+        private static void ReadAllRegistersWithDetails(Mcp25xxx mcp25xxx)
+        {
+            Console.WriteLine("Read All Registers");
+            Console.WriteLine();
+            ReadAllMessageTransmitRegistersWithDetails(mcp25xxx);
+            ReadAllMessageReceiveRegistersWithDetails(mcp25xxx);
+            ReadAllAcceptanceFilterRegistersWithDetails(mcp25xxx);
+            ReadAllBitTimeConfigurationRegistersWithDetails(mcp25xxx);
+            ReadAllErrorDetectionRegistersWithDetails(mcp25xxx);
+            ReadAllInterruptRegistersWithDetails(mcp25xxx);
+            ReadAllCanControlRegistersWithDetails(mcp25xxx);
+        }
+
+        private static byte ConsoleWriteRegisterAddressDetails(Mcp25xxx mcp25xxx, Address address)
+        {
+            byte value = mcp25xxx.Read(address);
+            Console.WriteLine($"  0x{(byte)address:X2} - {address}: 0x{value:X2}");
+            return value;
+        }
+
+        private static void ConsoleWriteRegisterItemDetails(string name, string details)
+        {
+            Console.WriteLine($"{name,15}: {details}");
+        }
+
+        private static void ReadAllMessageTransmitRegistersWithDetails(Mcp25xxx mcp25xxx)
+        {
+            Console.WriteLine("Message Transmit Registers");
+
+            var txRtsCtrl = new TxRtsCtrl(ConsoleWriteRegisterAddressDetails(mcp25xxx, Address.TxRtsCtrl));
```

Comment:
could you somehow do this with reflection (i.e. typeof(enumtype).GetValues or something along these lines)

Reply:
Honestly, I didn't think this was efficient and believe this part of the sample could be removed later.  It was more of a helper to troubleshoot while developing.  I have other thoughts on how we can display binding data (registers, memory, pins, etc.) and do it (hopefully) more consistently.  But I'm still pondering this and not at a point to create a proposal yet.

The approach I took on this was to show the value of the register followed by the respective bits.  While this is a must, I think trying to do the reflection method would return values in a different order (but not sure).

I'll like to hold off on this for now.

Reply:
@shaggygi I was doing same thing when started on the RFID stuff - although it was like 3 lines of code (foreach on all enum values and Console.WriteLine)

Reply:
Okay.  I'll try to review tomorrow to see if it can be cleaned up a bit.

Reply:
@krwq I was able to simplify some of the sample code based on your comments regarding reflection for the properties.  I don't want to do this for the actual registers as the current code groups them for displaying.  I'll commit the code tomorrow afternoon for the property updates.